libcodechecker/server/client_db_access_server.py

```diff
@@ -266,31 +535,43 @@ class CCSimpleHttpServer(HTTPServer):
     def __init__(self,
                  server_address,
                  RequestHandlerClass,
-                 db_conn_string,
+                 config_directory,
+                 product_db_sql_server,
                  pckg_data,
                  suppress_handler,
-                 db_version_info,
+                 context,
+                 check_env,
                  manager):
 
         LOG.debug("Initializing HTTP server...")
 
+        self.config_directory = config_directory
         self.www_root = pckg_data['www_root']
         self.doc_root = pckg_data['doc_root']
         self.checker_md_docs = pckg_data['checker_md_docs']
         self.checker_md_docs_map = pckg_data['checker_md_docs_map']
         self.version = pckg_data['version']
         self.suppress_handler = suppress_handler
-        self.db_version_info = db_version_info
-        self.__engine = database_handler.SQLServer.create_engine(
-            db_conn_string)
-
-        Session = scoped_session(sessionmaker())
-        Session.configure(bind=self.__engine)
-        self.sc_session = Session
+        self.context = context
+        self.check_env = check_env
```

Comment:
They store the external environment "pointers" from `libhandlers/server.py`, where this class is theoretically initialised. Otherwise, at **every** product database connection, the context configuration files and the checker environment would need to be loaded, which results in an increased disk usage, and an increased bloat in the log output.